### DISCUSSION TITLE
One Longhorn RWX volume with multiple PVCs

### DISCUSSION THREAD

**Discussion by excieve** [longhorn/longhorn#3591] [Q&A] [answered]
As an experiment I'm trying to set up a cluster-wide RWX volume. Since PVCs are namespaced and there's a 1-1 relationship between the PVCs and PVs my theory was to manually create a LH volume with RWX mode and then create multiple PVs and PVCs in different namespaces referencing this same LH volume by handle.

Kubernetes did not complain on this and everything was successfully bound, however when I actually attached one of those PVCs to a pod the Longhorn volume started detaching/attaching in a loop, making the mount unusable in the pod (even if other PVCs were not attached).

This behaviour doesn't happen when there's just one PV/PVC referencing a LH volume. So I wonder if that's something expected to happen? Is there a limitation in Longhorn as Kubernetes seems to be technically content with having such a setup?

**Comment by joshimoo** [comment, marked as answer]
**This is unsupported and generally a bad Idea even on other kubernetes volumes**
For example:
- what happens if you extend (PVC/PV)1
- what happens if you delete (PVC/PV)1

**Buyer beware, this is just a random though from an engineer**
If this is just a toy example, to learn more about kubernetes/storage.
What you can do is deploy a deamonset utilizing a longhorn RWX volume.
Then use kubernetes native NFS pvc's pointing to the cluster ip service of the longhorn RWX volume.
One would need to treat the LH PVC as a master so all administrative actions would need to happen against it.

Further exploration can be done by utilizing the traditional provisioner example, https://longhorn.io/docs/1.0.2/advanced-resources/rwx-workloads/ then modifying the provisioner code to fit your use case :)

**Reply by excieve** [reply]
Thanks for this @joshimoo! I expected that it's not supported, but native NFS is an interesting approach.
